python/ml4ir/base/model/scoring/interaction_model.py

```diff
@@ -35,9 +35,13 @@ def __call__(self, inputs: Dict[str, Input]):
             computing custom metrics and losses
         """
         train_features, metadata_features = self.feature_layer_op(inputs)
-        train_features, metadata_features = self.transform_features_op(
-            train_features, metadata_features
-        )
+        # separate the default learnable features from the fr_features
+        if self.feature_config.feature_exists('fr', trainable=True):
+            train_fr_features, _ = self.transform_features_op({'fr_feature':train_features['fr_feature']}, metadata_features)
```

Comment:
@mohazahran I have a doubt that this does what you expect it to do. You define a preprocessing function and are then trying to use it from the interaction_model. The interaction_model requires functions to be specified in the `feature_layer_info` part of the config YAML.

Reply:
I believe it should work because the way I set it up is the pre-processing function converts fr to one-hot encoded vector for training and mask the fr for validation and testing. So the input feature to the mode is already processed and don't need any further intervention.